src/main/kotlin/com/terraformation/backend/tracking/model/NotificationCriteriaModel.kt

```diff
@@ -0,0 +1,48 @@
+package com.terraformation.backend.tracking.model
+
+import com.terraformation.backend.db.tracking.PlantingSiteId
+import com.terraformation.backend.db.tracking.tables.records.PlantingSitesRecord
+import com.terraformation.backend.db.tracking.tables.references.PLANTING_SITES
+import com.terraformation.backend.tracking.event.ObservationSchedulingNotificationEvent
+import com.terraformation.backend.tracking.event.ScheduleObservationNotificationEvent
+import com.terraformation.backend.tracking.event.ScheduleObservationReminderNotificationEvent
+import java.time.Instant
+import org.jooq.Condition
+import org.jooq.TableField
+import org.jooq.impl.DSL
+
+class NotificationCriteriaModel {
```

Comment:
Structurally this seems reasonable. Maybe shorten the names:

* `NotificationCriteriaModel` -> `NotificationCriteria` since it's not really a model class, just a namespace.
* `ObservationSchedulingNotifications` -> `ObservationScheduling` since "notification" is already implicit from the outer class.